crates/swc_ecma_transforms_typescript/src/transform.rs

```diff
@@ -206,439 +339,316 @@ impl VisitMut for Transform {
-            .into();
         }
         node.visit_mut_children_with(self);
     }
 
-    fn visit_mut_for_stmt(&mut self, node: &mut ForStmt) {
-        if stmt_is_enum(&node.body) {
-            node.body = BlockStmt {
-                stmts: vec![*node.body.take()],
-                ..Default::default()
-            }
-            .into();
-        }
+    fn visit_mut_prop(&mut self, node: &mut Prop) {
         node.visit_mut_children_with(self);
-    }
 
-    fn visit_mut_while_stmt(&mut self, node: &mut WhileStmt) {
-        if stmt_is_enum(&node.body) {
-            node.body = BlockStmt {
-                stmts: vec![*node.body.take()],
-                ..Default::default()
-            }
-            .into();
+        if let Some(ref_rewriter) = self.ref_rewriter.as_mut() {
+            ref_rewriter.exit_prop(node);
         }
-        node.visit_mut_children_with(self);
     }
 
-    fn visit_mut_do_while_stmt(&mut self, node: &mut DoWhileStmt) {
-        if stmt_is_enum(&node.body) {
-            node.body = BlockStmt {
-                stmts: vec![*node.body.take()],
-                ..Default::default()
-            }
-            .into();
-        }
-        node.visit_mut_children_with(self);
+    fn visit_mut_var_declarator(&mut self, n: &mut VarDeclarator) {
+        let ref_rewriter = self.ref_rewriter.take();
+        n.name.visit_mut_with(self);
+        self.ref_rewriter = ref_rewriter;
+        n.init.visit_mut_with(self);
     }
 
-    fn visit_mut_labeled_stmt(&mut self, node: &mut LabeledStmt) {
-        if stmt_is_enum(&node.body) {
-            node.body = BlockStmt {
-                stmts: vec![*node.body.take()],
-                ..Default::default()
-            }
-            .into();
-        }
+    fn visit_mut_pat(&mut self, node: &mut Pat) {
         node.visit_mut_children_with(self);
-    }
 
-    fn visit_mut_with_stmt(&mut self, node: &mut WithStmt) {
-        if stmt_is_enum(&node.body) {
-            node.body = BlockStmt {
-                stmts: vec![*node.body.take()],
-                ..Default::default()
-            }
-            .into();
+        if let Some(ref_rewriter) = self.ref_rewriter.as_mut() {
+            ref_rewriter.exit_pat(node);
         }
-        node.visit_mut_children_with(self);
     }
-}
 
-impl Transform {
-    fn visit_mut_stmts_for_enum_or_module(&mut self, n: &mut Vec<Stmt>) {
-        let found = (**n).iter().any(|stmt| stmt.is_enum_or_module());
+    fn visit_mut_expr(&mut self, node: &mut Expr) {
+        self.enter_expr_for_inline_enum(node);
 
-        if !found {
-            return;
+        maybe_grow_default(|| node.visit_mut_children_with(self));
+
+        if let Some(ref_rewriter) = self.ref_rewriter.as_mut() {
+            ref_rewriter.exit_expr(node);
         }
+    }
 
-        let mut decl_id_record = AHashSet::<Id>::default();
+    fn visit_mut_assign_expr(&mut self, n: &mut AssignExpr) {
+        let is_lhs = mem::replace(&mut self.is_lhs, true);
```

Comment:
It will be helpful if we have semantic analysis.

https://github.com/oxc-project/oxc/blob/02fedf5123e10b9723b17348aa56afabc69b66dd/crates/oxc_syntax/src/reference.rs#L76-L93